app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -138,6 +141,18 @@ class WireActivityViewModel @Inject constructor(
                 .distinctUntilChanged()
                 .collect { _observeSyncFlowState.emit(it) }
         }
+        viewModelScope.launch(dispatchers.io()) {
+            observeIfAppUpdateRequired(BuildConfig.VERSION_CODE)
+                .distinctUntilChanged()
+                .map {
+                    // if update required, need to make sure that it's actually available
+                    if (it) appUpdateManager.isAppUpdateAvailable()
```

Comment:
Hmm, if update is required but not yet available, I think we should still prevent the user from using the app, because this version is blacklisted, at least I understand it this way 😄

Reply:
hmmm, that is the question for the discuss 🤔